torchdata/dataloader2/communication/eventloop.py

```diff
@@ -26,16 +29,57 @@
 
 __all__ = [
     "DataPipeToQueuesLoop",
-    "SpawnProcessForDataPipeline",
-    "SpawnThreadForDataPipeline",
+    "CreateProcessForDataPipeline",
+    "CreateProcessForMultipleDataPipelines",
+    "CreateThreadForDataPipeline",
 ]
 
 
+def MultipleDataPipesToQueuesLoop(source_datapipes, req_queues, res_queues, call_on_process_init=None):
+    r"""
+    Set the appropriate pipes and protocol server type, and create a loop over multiple datapipes
+    with the protocol server in a non-blocking manner.
+    """
+    assert call_on_process_init is None, "``MultipleDataPipesToQueuesLoop`` does not support call_on_process_init"
+    assert len(source_datapipes) == len(req_queues) and len(req_queues) == len(
+        res_queues
+    ), "``MultipleDataPipesToQueuesLoop`` requires the same number of datapipes, request queues and response queues"
+
+    torch.set_num_threads(1)
```

Comment:
noob question: what's this for?

Reply:
IIRC, this was introduced to disable `OpenMP` in dataloader workers. This is because OpenMP would create number of threads that equals to the number of CPU cores by default. And, with multiprocessing enabled, `num_workers x num_threads_per_worker` threads will be created. This won't provide any further benefit.

Besides, `OpenMP` features should not be enabled if any `OpenMP` features are utilized in the main process and before subprocesses are forked.

Any suggestion?